Classes/URKArchive.mm

```diff
@@ -331,7 +403,32 @@ - (nullable NSURL *)firstVolumeURL {
 
 - (nullable NSArray<NSString*> *)listVolumePaths:(NSError **)error
 {
-    return @[];
+    __block NSMutableArray<NSString*> *volumePaths = [NSMutableArray new];
+    
+    URKArchive * dummy = self;
+    NSURL * firstVolumeURL = [self firstVolumeURL];
+    if (firstVolumeURL != self.fileURL)
+    {
+        NSError *firstVolumeURLError = nil;
+        
+        dummy = [self initWithURL:firstVolumeURL error:&firstVolumeURLError];
+        if (firstVolumeURLError != nil && error) {
+            *error = firstVolumeURLError;
+            return nil;
+        }
+    }
+    
+    NSArray<URKFileInfo*> *listFileInfo = [dummy listFileInfo:error];
+    
+    if (listFileInfo == nil)
+        return nil;
+    
+    for (URKFileInfo* info in listFileInfo) {
+        if (![volumePaths containsObject:info.archiveName])
```

Comment:
Would check it out in two weeks. Will be busy until then. Let me know if there's something else to change :)